src/utils.js

```diff
@@ -0,0 +1,45 @@
+const axios = require('axios');
+const sharp = require('sharp');
+const Gifsicle = require('gifsicle-stream');
+
+const logger = require('./logger');
+
+exports.fileTypeIsSupported = async metadata => {
+  const extensions = ['png', 'jpeg', 'jpg', 'gif'];
+
+  if (!extensions.includes(metadata.file.filetype)) {
+    const msg = `Unsupported filetype ${metadata.file.filetype}`;
+
+    logger.error(msg);
+
+    throw new Error(msg);
+  }
+};
+
+exports.streamAsPromise = stream => {
+  return new Promise((resolve, reject) =>
+    stream.on('finish', resolve).on('error', reject)
+  );
+};
+
+exports.getResizer = mimetype => {
+  return mimetype === 'image/gif'
+    ? new Gifsicle(['--resize-fit', '128'])
+    : sharp()
+        .max()
+        .resize(128, 128, {
+          fit: sharp.fit.inside,
+          withoutEnlargement: true,
+        });
+};
+
+exports.downloadAsStream = url => {
```

Comment:
Is this a stream or a promise? In the test you have `response.then...` which suggests promise?

Reply:
Axios returns a promise, but the `result.data` (result is what is passed into the then of the promise) is a readable stream.

I could handle the response and return only `result.data` within `downloadAsStream` which would probably make more sense?